massa-grpc/src/public.rs

```diff
@@ -482,33 +484,34 @@ pub(crate) fn get_operations(
         })
         .collect::<Result<_, _>>()?;
 
-    let read_blocks = grpc.storage.read_blocks();
-    let read_ops = grpc.storage.read_operations();
-
-    // Get the operations and the list of blocks that contain them from storage
-    let storage_info: Vec<(&SecureShareOperation, HashSet<BlockId>)> = operation_ids
-        .into_iter()
-        .filter_map(|ope_id| {
-            read_ops.get(&ope_id).map(|secure_share| {
-                let block_ids = read_blocks
-                    .get_blocks_by_operation(&ope_id)
-                    .map(|hashset| hashset.iter().cloned().collect::<HashSet<BlockId>>())
-                    .unwrap_or_default();
-
-                (secure_share, block_ids)
+    let storage_info: Vec<(SecureShareOperation, PreHashSet<BlockId>)> = {
+        let read_blocks = grpc.storage.read_blocks();
+        let read_ops = grpc.storage.read_operations();
```

Comment:
never take the two at the same time !

start by locking the ops, list them, then unlock, then lock blocks and add get_blocks_by_operation information

Reply:
We should do the same in the old API

Reply:
yes, can you make the changes directly in this PR ?

Reply:
In JsonRPC API, we have a different behaviour, we took the lock for every operation, release it. 

What's more efficient ?